lib/hammer_cli/options/option_definition.rb

```diff
@@ -104,20 +100,23 @@ def value_description
       end
 
       def default_conversion_block
-        if !value_formatter.nil?
-          value_formatter.method(:format)
-        elsif flag?
+        if flag?
           Clamp.method(:truthy?)
+        else
+          formatter = value_formatter.method(:format)
+          lambda do |value|
+            if value == (ENV['HAMMER_NIL'] || HammerCLI::Options::NIL_SUBST)
```

Comment:
should we consider empty string as nil value? I think users would tend to do `export HAMMER_NIL=some` and they would need to do `unset` but mistake such as `HAMMER_NIL= hammer user list` would mark empty strings as nil?